buildconfig/config_msys2.py

```diff
@@ -16,6 +16,11 @@
 from glob import glob
 from distutils.sysconfig import get_python_inc
 
+try:
+    BuildError
+except NameError:
+    BuildError = TypeError
```

Comment:
what is this for? Why is this here and in some other files?

Reply:
It is an _undefined name_ which has the potential to raise `NameError` at runtime so let's avoid that.

Reply:
Again... the codepath that actually calls `BuildError` is super unlikely to run, and even if it does, I think it's better if it errored with `RuntimeError` directly in code instead of dealing with `BuildError` aliases which is not even present elsewhere in the code

Reply:
Nice, I see you fixed this. But you are raising `TypeError` there which seems to be the wrong error to raise here, ideally it should be a `ValueError` (because the value should have been 32 or 64) or a `RuntimeError`